src/wp-includes/media-template.php

```diff
@@ -357,8 +357,23 @@ function wp_print_media_templates() {
 			<div class="thumbnail thumbnail-{{ data.type }}">
 				<# if ( data.uploading ) { #>
 					<div class="media-progress-bar"><div></div></div>
-				<# } else if ( data.sizes && data.sizes.large ) { #>
-					<img class="details-image" src="{{ data.sizes.large.url }}" draggable="false" alt="" />
+				<# } else if ( data.sizes && data.sizes.large ) {
+					var large_image_preview_url = data.sizes.large.url;
+					if ( data.sizes.full ) {
+						var full_image_crop_id = data.sizes.full.url.match(/^.+-(e[0-9]{13})\.[^\\/]+$/)?.[1],
```

Comment:
I hope it's OK to use optional chaining operator here (`?.`).
According to [this page](https://make.wordpress.org/core/handbook/best-practices/browser-support/) all browsers supported by WordPress should support this syntax.

Reply:
@anton-vlasenko this regex is run in PHP, right? so the PHP engine is what needs to support the optional chaining. which, I think it does for versions we support. Adding some test cases that get run across PHP versions would help ensure that

Reply:
@adamsilverstein 

> this regex is run in PHP, right? 

This regex is [run in JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match).
According to [this page (open the `Date relative` tab)](https://make.wordpress.org/core/handbook/best-practices/browser-support/), **the vast majority** of the browsers released in 2020 and later support this syntax.

Reply:
that covers all our supported browsers so should be fine. what are the consequences if it fails? console error?